src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -405,14 +405,21 @@ public static IntPtr GetCurrentCalleeOpenStaticDelegateFunctionPointer()
             return PInvokeMarshal.GetCurrentCalleeDelegate<T>();
         }
 
-        public static IntPtr ConvertManagedComInterfaceToNative(object pUnk)
+        public static IntPtr ConvertManagedComInterfaceToNative(object pUnk, Type convertToType)
         {
             if (pUnk == null)
             {
                 return IntPtr.Zero;
             }
 
+#if TARGET_WINDOWS
+#pragma warning disable CA1416
+            Guid interfaceGuid = convertToType.GUID;
```

Comment:
It would be more inline with the NativeAOT design printiples for the GUID to be passed as an argument into this method by the toolchain instead of computing it at runtime via reflection.

@MichalStrehovsky Do you expect that there are going to be special measures required to make the interface reflectable going forward, or is this going to just work?

Reply:
Using reflection to fetch the GUID here is also going to make the marshalling much slower compared to regular CLR.

Reply:
I think the design we will have to settle with is that _types_ are always reflectable, so fetching a GUID should be reliable, but yes, this is not going to be very fast.

It shouldn't be too much of a problem to change this signature of this helper to take an actual GUID, and in the compiler:

* Use `.GetDecodedCustomAttributes` to get the attributes on the type.
* Parse the string into a GUID.
* Emit IL to call the `Guid(Int32, Int16, Int16, Byte, Byte, Byte, Byte, Byte, Byte, Byte, Byte)` constructor with the parsed GUID values.

We can scope out types that don't have a GuidAttribute for now (runtime reflection would not compute the correct GUID for them either).

Reply:
@MichalStrehovsky I'm not sure that I properly understand how to jump from `TypeDesc` to `EcmaType`.
I plan to use following 
```
CustomAttributeValue<TypeDesc>? customAttributeValue = ((EcmaType)this.ManagedParameterType.GetTypeDefinition()).GetDecodedCustomAttribute("", "GuidAttribute");
```

Reply:
`(this.ManagedParameterType as EcmaType)?.GetDecodedCustomAttribute("System.Runtime.InteropServices", "GuidAttribute")`

- The attribute namespace has to be specified
- `GetTypeDefinition` would be only needed for generic types. Built-in COM interop does not work for generic types so you can just skip that part.